numpy/core/tests/test_cpu_features.py

```diff
@@ -48,11 +57,72 @@ def assert_features_equal(actual, desired, fname):
         "%s"
     ) % (fname, actual, desired, error_report))
 
+def _text_to_list(txt):
+    out = txt.strip("][\n").replace("'", "").split(', ')
+    return None if out[0] == "" else out
+
 class AbstractTest:
     features = []
     features_groups = {}
     features_map = {}
     features_flags = set()
+    cwd = pathlib.Path(__file__).parent.resolve()
+    env = os.environ
+    _enable = os.environ.pop('NPY_ENABLE_CPU_FEATURES', None)
+    _disable = os.environ.pop('NPY_DISABLE_CPU_FEATURES', None)
+    SUBPROCESS_ARGS = dict(cwd=cwd, capture_output=True, text=True, check=True)
+    unavailable_feats = [
+        feat for feat in __cpu_dispatch__ if not __cpu_features__[feat]
+    ]
+    UNAVAILABLE_FEAT = (
+        None if len(unavailable_feats) == 0
+        else unavailable_feats[0]
+    )
+    BASELINE_FEAT = None if len(__cpu_baseline__) == 0 else __cpu_baseline__[0]
+
+    def _run(self):
+        path_to_np = os.path.dirname(np.__file__)
+        return subprocess.run(
+            ['python', f'{path_to_np}/core/tests/_runtime_detect.py'],
```

Comment:
```suggestion
            [sys.executable, f'{path_to_np}/core/tests/_runtime_detect.py'],
```
Just tried to have a look locally on my linux machine and it barfed on this.  Probably not all to it, but maybe it helps.